src/theory/trust_substitutions.cpp

```diff
@@ -215,35 +227,34 @@ TrustNode TrustSubstitutionMap::apply(Node n, bool doRewrite)
   // notice this proof is reused.
   d_applyPg->addSteps(*d_tspb.get());
   d_tspb->clear();
-  return TrustNode::mkTrustRewrite(n, ns, d_applyPg.get());
+  Trace("trust-subs-pf") << "...finish, make proof" << std::endl;
+  return d_applyPg->getProofFor(eq);
 }
 
+std::string TrustSubstitutionMap::identify() const { return d_name; }
+
 SubstitutionMap& TrustSubstitutionMap::get() { return d_subs; }
 
 bool TrustSubstitutionMap::isProofEnabled() const
 {
   return d_subsPg != nullptr;
 }
 
-Node TrustSubstitutionMap::getCurrentSubstitution()
+Node TrustSubstitutionMap::getSubstitution(size_t index)
 {
-  Assert(isProofEnabled());
-  if (!d_currentSubs.get().isNull())
-  {
-    return d_currentSubs;
-  }
+  Assert(index <= d_tsubs.size());
   std::vector<Node> csubsChildren;
-  for (const TrustNode& tns : d_tsubs)
+  for (size_t i = 0; i < index; i++)
   {
-    csubsChildren.push_back(tns.getProven());
+    csubsChildren.push_back(d_tsubs[i].getProven());
   }
   std::reverse(csubsChildren.begin(),csubsChildren.end());
-  d_currentSubs = NodeManager::currentNM()->mkAnd(csubsChildren);
-  if (d_currentSubs.get().getKind() == kind::AND)
+  Node cs = NodeManager::currentNM()->mkAnd(csubsChildren);
+  if (cs.getKind() == kind::AND)
   {
-    d_subsPg->addStep(d_currentSubs, PfRule::AND_INTRO, csubsChildren, {});
+    d_subsPg->addStep(cs, PfRule::AND_INTRO, csubsChildren, {});
   }
```

Comment:
Since you have the size you could compute this directly rather than check kinds and vector size, if you think it'd matter.